### ISSUE TITLE
Changing the default appicon doesn't work.

### ISSUE THREAD

**Issue by gfmoore** [dotnet/maui#7786]
### Description

- [ ] https://github.com/xamarin/xamarin-macios/issues/12790
- [ ] https://github.com/xamarin/xamarin-android/issues/7147

Please see this issue on the documentation https://github.com/dotnet/docs-maui/issues/563 that I raised.

I have an svg icon called bicycle.svg
![bicycle](https://user-images.githubusercontent.com/22768967/172150131-86c95eba-2ca1-48ab-9d75-1faa8b6568de.svg)

I originally followed the advice in the article https://docs.microsoft.com/en-us/dotnet/maui/user-interface/images/app-icons?tabs=android 

This didn't work. The app icon remained unchanged (in the Pixel 5 emulator). I followed the advice to clean, build, rebuild, debug, dotnet clean from command line no change. 

During the process of trying to get things working I got all sorts of error messages, some referring to the icon being locked. So rebooted, but sometimes still locked. dotnet clean seemed to work better than other approaches. 

 However, my situation is that now I have the following:
![image](https://user-images.githubusercontent.com/22768967/172150852-9644317c-15f9-45cd-ad47-b625e4e966a8.png)

So I have my icon called bicycle.svg in the Resources/AppIcon folder, the .csproj file is referencing that file, however my app will now not build

![image](https://user-images.githubusercontent.com/22768967/172151184-00d7ade6-91e9-43a1-899c-bcf51314eb35.png)

I dragged the bicycle.icon back into the folder (on visual studio and replaced the existing one. That failed again. I had suspected that dragging the required icon into the folder would update other manifests but it seems not.

I renamed my bicycle.svg icon to appicon, removed the one in the Resource/AppIcon folder

I noted that deleting the icon from the visual studio folder also removed the App Icon 
<MauiIcon Include="Resources\AppIcon\bicycle.svg" Color="#000000" /> line in the .csproj

but adding the appicon.svg did not restore the line in .csproj

I had to manually restore these lines

Once I got the app working again, the app icon in the emulator still hadn't changed.

 So questions:
1) Does the name of the appicon have to appicon.svg? I don't see why it cannot be anyname.svg as long as it matches the .csproj

2) In order to update the manifests in android etc (which is what I suspect is not happening), what is the correct procedure. Is there a command I can run to ensure this happens.

Here is the appicon in the emulator drawer
![image](https://user-images.githubusercontent.com/22768967/172153452-833ae951-8cf4-4b1c-a7be-7c9512596058.png)

The one to the left is a test app with no changes to the appicon. The TSDZ2... is the one that I am trying to be the bicycle and it looks like an android icon?

I'm sorry this question is a little rambling, but I've had all sorts of issues and I couldn't figure out how best to describe what happens. The upshot seems to be that the process as described in the docs doesn't work.

 

### Steps to Reproduce

As above

### Version with bug

6.0 Release Candidate 3

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Pixel 5 API 30

### Did you find any workaround?

No

**Comment by janseris** [comment]
Tip:
for real clean, delete bin and debug folders (the Clean Solution or Clean project doesn't really clean a MAUI project build output).
These are my discoveries for app icon and splash screen:

![image](https://user-images.githubusercontent.com/64279914/172160700-7184b6f0-1c6b-4570-a609-30e5822dbe72.png)

**Comment by gfmoore** [comment]
Thanks for this. I'll check it out ASAP. 😀

**Comment by jfversluis** [comment]
Adding needs-info for now so we wait on your report back. Additionally it's probably a duplicate of one of the issues that is already opened then :)

**Comment by adegeo** [comment]
I would vote to have this issue track all the others..

**Comment by VincentBu** [comment]
I'm using vs main build 32609.239 and [janseris](https://github.com/janseris)'s workaround works for me.

**Comment by gfmoore** [comment]
Well I've nearly got something. I'm still using the svg because since the original appicon appiconfg are svg and they should rescale? But I take the point that png may be better.

So on my actual phone I now have my black background, my bicycle icon, but it has .NET overlayed on it in white.

![image](https://user-images.githubusercontent.com/22768967/173050672-56bcfd9c-6a05-4597-a33a-4abe62883cff.png)

Here's my appicon.svg
![appicon](https://user-images.githubusercontent.com/22768967/173050757-ac250734-6209-432c-bbeb-25f46262100a.svg)

My appiconfg.svg (Which by the way doesn't appear under the resources AppIcon folder
![appiconfg](https://user-images.githubusercontent.com/22768967/173050791-6cfabdbc-2b4d-44d2-821f-60f740c9427b.svg)

And here is the .csproj
```
	<ItemGroup>
		<!-- App Icon -->
                <MauiIcon Include="Resources\AppIcon\appicon.svg" ForegroundFile="Resources\AppIcon\appiconfg.svg" Color="#000000" />
    
		<!-- Splash Screen -->
                <MauiSplashScreen Include="Resources\Splash\bicycle.svg" Color="#000000" />
```
 
(By the way the spashscreen has stopped resizing to fit within the device window - it has done before, but was flaky)

And just to add salt to the wound here is Mandy Montequila extolling how magical using app icons is:
https://mybuild.microsoft.com/en-US/sessions/802e54dd-5b4a-460f-8e74-240e4a78f9c9

at 5:38

And she's a senior pm at Microsoft .Maui !!!! (I refrained from rude comments.)

**Comment by gfmoore** [comment]
oops now I've posted it I can see where the .Net is coming from - doh, when I looked it was blank. Sorry I'll edit it now.

**Comment by gfmoore** [comment]
Ahh it sort of works now
Here is the appiconfg.svg I used Inkscape to edit and it was a white lettering on the default white background - doh.
![appiconfg](https://user-images.githubusercontent.com/22768967/173052421-970b9388-f921-4477-b5ed-7dbb649e0e9d.svg)

I note that the image of the bike is a little larger than I expected - similar non rescaling issue as per the splashscreen?

Thanks to janseris for his tips.

**Comment by janseris** [comment]
@gfmoore well you know PMs are to trade products not to guarantee their correct functionality.
Which is sad that with my loved C#/.NET, the MAUI (worse product state) and Razor tooling team (better product state) is just marketing broken products.

**Comment by gfmoore** [comment]
Being a naive and trusting person I expected this to all work. Oh wait, since Microsoft started and I used dos on the IBM PC clones Microsoft have always released buggy, malfunctioning products and expected the community to report errors and fix it for them. Nothing has changed in the last 40 years (man how much of my life have I wasted fixing bugs!!!! Oh dear.) :(

Man it's only an app icon, how hard is it to test and get it right and now we are on GA.

**Comment by gfmoore** [comment]
Sadly I suspect this will never get assigned to anyone with any skill as now we are at GA, MS will disband the team except for some lowly maintainers and move on to the next big thing, because as far as they are concerned it's all done and dusted. And don't get me started on lack of Bluetooth functionality...

**Comment by gfmoore** [comment]
Screaming: Now my app won't build or deploy
![image](https://user-images.githubusercontent.com/22768967/173056488-99b52e09-84c6-4646-b2a8-9e0b23467cb0.png)

**Comment by gfmoore** [comment]
I had to replace my edited appiconfg.svg file with an unedited one from another app!

**Comment by janseris** [comment]
> I'm using vs main build 32609.239 and [janseris](https://github.com/janseris)'s workaround works for me.

Are issues in maui repo also fixed or only verified?

**Comment by gfmoore** [comment]
So I've made progress and discoveries.
First the splash scrren not resizing - the spash scrren file needs to be called ... splash.svg!!! 

Had a great ide for the app icon. I just copied the appicon.svg and renamed it appiconfg.svg. So now my appicon overlays my appicon. Had all sorts of weird issues, so closed down Visual Studio, deleted the bin file, deleted the obj.debug file ran dotnet clean in my solution folder and restarted visual studio. Yippee I now have an app icon. I consider it is too big, but I can live with it for now.

So Microsoft when will you make the magic happen? When will you review the code and approach to this issue and make it work.

It should work with ANY svg name out of the box, it shouldn't cause errors just because files are changed or renamed (even if manual changes to .csproj have to be done, and certainly I shouldn't have to jump through hoops and delete folders etc just to get an app icon and splash screen to work!!!! Should I! Should I?

**Comment by gemma1987** [comment]
For me everything is working with the appicon, but it doesn't rescale, so now I have this on the maui app:
![image](https://user-images.githubusercontent.com/77026023/173236127-7a7c5ada-8438-4271-8485-1927ec480925.png)
And this in the xamarin.forms app:
![image](https://user-images.githubusercontent.com/77026023/173236141-f99f764f-bedc-4006-a744-2a3776495236.png)

As you can see I can change the name of the icon and place it in the resources file, but the app icon doesn't change size, whatever I try:

![image](https://user-images.githubusercontent.com/77026023/173236193-0bfad156-4b12-4346-ad58-49efda71a6ee.png)

**Comment by janseris** [comment]
> For me everything is working with the appicon, but it doesn't rescale, so now I have this on the maui app: ![image](https://user-images.githubusercontent.com/77026023/173236127-7a7c5ada-8438-4271-8485-1927ec480925.png) And this in the xamarin.forms app: ![image](https://user-images.githubusercontent.com/77026023/173236141-f99f764f-bedc-4006-a744-2a3776495236.png)
> 
> As you can see I can change the name of the icon and place it in the resources file, but the app icon doesn't change size, whatever I try:
> 
> ![image](https://user-images.githubusercontent.com/77026023/173236193-0bfad156-4b12-4346-ad58-49efda71a6ee.png)

https://github.com/dotnet/maui/issues/7305

**Comment by gemma1987** [comment]
> > For me everything is working with the appicon, but it doesn't rescale, so now I have this on the maui app: ![image](https://user-images.githubusercontent.com/77026023/173236127-7a7c5ada-8438-4271-8485-1927ec480925.png) And this in the xamarin.forms app: ![image](https://user-images.githubusercontent.com/77026023/173236141-f99f764f-bedc-4006-a744-2a3776495236.png)
> > As you can see I can change the name of the icon and place it in the resources file, but the app icon doesn't change size, whatever I try:
> > ![image](https://user-images.githubusercontent.com/77026023/173236193-0bfad156-4b12-4346-ad58-49efda71a6ee.png)
> 
> #7305

@janseris But the issue is not in the splashscreen (This is working perfectly fine) , it is only in the app icon, is this issue (#7305) connected to the sizing of the app icon? That's my only issue

**Comment by mattleibow** [comment]
I opened a new issue on the docs page about the missing section on actually using the new images provided : https://github.com/dotnet/docs-maui/issues/611

Also, this is being improved here: https://github.com/dotnet/maui/pull/7968

I realize there are a bunch of issues here and I am looking at them each to fix them all and make sure any docs/defaults/generation is happening correctly.

**Comment by gfmoore** [comment]
You may want to rename your icon.svg to appicon.svg. I got loads of error and warnings otherwise on build. Same with splash.svg. Obviously it can work, but didn't for me.

**Comment by mattleibow** [comment]
You need to make sure you update the values in the manifests to match the icon you are generating: 

https://github.com/dotnet/maui/blob/release/6.0.4xx/src/Templates/src/templates/maui-mobile/Platforms/Android/AndroidManifest.xml#L3

![image](https://user-images.githubusercontent.com/1096616/173241954-57e75ba5-6fac-4018-a33b-8f84075fc481.png)

**Comment by gfmoore** [comment]
That's interesting, thought it was supposed to be magic! 🤣

**Comment by mattleibow** [comment]
The other issue is that having the same name for the splash and icon causes issues because android sees those files and then decides that because they are the same name they are the same file - mainly because it is a flat folder so the last one wins.

**Comment by mattleibow** [comment]
> That's interesting, thought it was supposed to be magic! 🤣

More magic is coming https://github.com/dotnet/maui/pull/7968

**Comment by mattleibow** [comment]
Created another issue to track Resizetizer issues https://github.com/dotnet/maui/issues/7970

Hopefully we get to full magic soon!

**Comment by ismasanchez** [comment]
> For me everything is working with the appicon, but it doesn't rescale, so now I have this on the maui app: ![image](https://user-images.githubusercontent.com/77026023/173236127-7a7c5ada-8438-4271-8485-1927ec480925.png) And this in the xamarin.forms app: ![image](https://user-images.githubusercontent.com/77026023/173236141-f99f764f-bedc-4006-a744-2a3776495236.png)
> 
> As you can see I can change the name of the icon and place it in the resources file, but the app icon doesn't change size, whatever I try:
> 
> ![image](https://user-images.githubusercontent.com/77026023/173236193-0bfad156-4b12-4346-ad58-49efda71a6ee.png)

As a temporary solution I had to do the following steps to make the icon and splash screen show correctly (at least in Android):
1) Design your icon as a png (also works with svg) with a Canvas of 400x400px and place your icon centered with a max dimension of 200x200 px
2) Save that png as appiconfg.png in AppIcon folder
3) Include default appicon.svg file in AppIcon folder, change color by editing .svg directly in VS in fill filed (width and height should be 456).
4) In your project .csproj file include the following:
		
		<MauiIcon Include="Resources\AppIcon\appicon.svg" ForegroundFile="Resources\AppIcon\appiconfg.png" />
	
		
		<MauiSplashScreen Include="Resources\AppIcon\appiconfg.png" Color="#whatevercolor" />
5) Clean Solution in VS and Rebuild (if it triggers some errors about Maui.Resizetizer.targets, when you press F5 it will go away).
Hope it gets resolved soon!

**Comment by mattleibow** [comment]
Closing this issue as it is a more of a "link to other issues" issue and many of the smaller ones have been fixed.

There is the one issue where changing some splash properties does not re-generate (#9803) and then other where aspect ratios are not preserved (#5676).
